pkg/services/sqlstore/migrator/dialect.go

```diff
@@ -48,6 +48,8 @@ type Dialect interface {
 	NoOpSql() string
 
 	IsUniqueConstraintViolation(err error) bool
+	IsDeadlock(err error) bool
+	GetErrorCode(err error) string
```

Comment:
I can't find any usage of this function. Was it added before `IsDeadlock` and not needed anymore?

Reply:
It is not needed for this specific case, I included it just because I thought it could be useful being there for the investigation of similar future issues.

Reply:
Since different datasources have different error codes its not really usable beyond testing one database. So I dont think we need it. The dialect function you added is a better suggestion for how people can solve this in the future :)

Reply:
I would say as `GetErrorCode` and `isThisError` are very similar but `GetErrorCode` is probably easier to understand from the name. So what about we keep `GetErrorCode`, remove the `isThisError` and use:
```
func (db *Mysql) IsUniqueConstraintViolation(err error) bool {
	return db.GetErrorCode(err) == mysqlerr.ER_DUP_ENTRY
}
```

Reply:
Since the errors for each database are of different type the `GetErrorCode ` returns string.
Therefore this condition would need casting but anyway I think it's better to get rid of `GetErrorCode` and maybe find an more appropriate name for `isThisError`.